src/sbsign/sbsign.c

```diff
@@ -254,6 +254,10 @@ static int verb_sign(int argc, char *argv[], void *userdata) {
         if (dstfd < 0)
                 return log_error_errno(r, "Failed to open temporary file: %m");
 
+        /* Make sure output file has same mode as input file. */
+        if (fchmod(dstfd, st.st_mode & 07777) < 0)
```

Comment:
why btw?

is this because open_tmpfile_linkable() masks off the group/world bits? but if so, maybe it would be better to do `fchmod_umask(dstfd, 0666)`? That would just mean you get the ouput output set to whatever the umask says, which seems ideal to me, and what most tools would expect?